docs/maintaining.md

```diff
@@ -92,3 +92,118 @@ made to always install the latest released version.
 
 Update the release branch by fast-forwarding it to the appropriate SHA (matching
 the latest tag) and pushing it back upstream.
+
+## `grafana/prometheus` Maintainence
+
+Grafana Labs is using the Agent for their internal monitoring and want to take
+advantage of the Agent to proof-of-concept additions to Prometheus before they
+get moved upstream. A `grafana/prometheus` repository is maintained by Grafana
+Labs where all non-trivial changes will go first. Doing this allows for getting
+Cortex-specific changes moving along faster and providing strong evidence
```

Comment:
Are most changes we're testing in the agent first really Cortex specific? I'd say:
> Testing these more experimental changes in a separate repo. that we can vendor into the agent allows us to iterate on them faster and...